main/core/src/EBox/RESTClient.pm

```diff
@@ -306,7 +308,7 @@ sub request {
     $ua->proxy('https', $ENV{HTTP_PROXY}) if (exists $ENV{HTTP_PROXY});
 
     my $req = HTTP::Request->new( $method => $self->{server} . $path );
-    if ( exists $self->{credentials} ) {
+    if (defined($self->{credentials})) {
```

Comment:
Maybe just assert self->{credentials} because is simpler and (secondary) also protects against empty string?